### PR TITLE
Add Studio parameter

### PR DESCRIPTION
**Pull request by delishad21**
Currently, the application is only able keep track of student classes using the "tag" functionality.

To allow for more restrictions to be placed on the structure of studio parameters and to pave the way for the implementation of future features, lets add an additional optional parameter of "Studio".

Code coverage was also added for the studio parameter, along with some additional ones for the matric parameter as well.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 1]
```diff
@@ -116,6 +136,14 @@ public void parse_repeatedNonTagValue_failure() {
         assertParseFailure(parser, INVALID_ADDRESS_DESC + validExpectedPersonString,
                 Messages.getErrorMessageForDuplicatePrefixes(PREFIX_ADDRESS));
 
+        // invalid matric number
+        assertParseFailure(parser, INVALID_MATRIC_DESC + validExpectedPersonString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_MATRIC_NUMBER));
+
```
Comment (Zer0Legion) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:142]: Thank you Johan for checking for duplicate Matric numbers as well!

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -208,4 +210,31 @@ public void parseMatric_validMatric_success() throws ParseException {
...
+    @Test
+    public void parseStudio_invalidStudio_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseStudio(INVALID_STUDIO));
+    }
+
+    @Test
+    public void parseStudio_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseStudio(null));
+    }
+
```
Comment (Zer0Legion) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:234]: Very insightful and comprehensive unit testing. Lets strive to maximize code coverage

**Review by Zer0Legion — COMMENTED** [review]
Very neat and readable code. Small issues raised by codecov bot but I am sure we can resolve it in the future. Overall great job Johan!

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 3]
```diff
@@ -208,4 +210,31 @@ public void parseMatric_validMatric_success() throws ParseException {
...
+    @Test
+    public void parseStudio_optionalFieldMissing_returnsEmptyStudio() throws ParseException {
+        seedu.address.model.student.Studio expected = new Studio("");
```
Comment (jayllo-c) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:237]: is `seedu.address...Studio` necessary or is simply using `Studio` enough?

File: src/main/java/seedu/address/logic/commands/AddCommand.java
Comment (jayllo-c) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:1]: Great that you've changed the commands to ensure that the parameter is incorporated into the functionality of the app properly

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java
Comment (jayllo-c) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:1]: Great catch ensuring that the new parameter should be stored in memory!

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java
Comment (jayllo-c) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:1]: Good job ensuring that the unit testing accounts for all the additions and changes made to the code!

**Review by jayllo-c — COMMENTED** [review]
Good job implementing the studio parameter. The code is neat, aligns with the matric parameter in the implementation, and is comprehensively tested!

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 4]
```diff
@@ -66,6 +69,14 @@ private static Matric handleOptionalMatric(ArgumentMultimap argMultimap) throws
         }
     }
 
+    private static Studio handleOptionalStudio(ArgumentMultimap argMultimap) throws ParseException {
```
Comment (Pughal77) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:72]: Nice abstraction to handle the optional parameter on its own method

File: src/main/java/seedu/address/model/student/Studio.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        checkArgument(isValidConstructorParam(studio), MESSAGE_CONSTRAINTS);
+        this.studio = studio;
+    }
+
+    /**
+     * Returns true if a given string is a valid studio for constructing a new Studio.
+     * @param test
+     * @return
+     */
+    static boolean isValidConstructorParam(String test) {
```
Comment (Pughal77) [inline src/main/java/seedu/address/model/student/Studio.java:32]: Useful to prevent any errors

**Review by Pughal77 — COMMENTED** [review]
Nice, clean implementation of a new field to Person. Great integration to the overall app

**Pull request merged**